graphql-server/src/utils/flattening.ts

```diff
@@ -0,0 +1,337 @@
+        cachedData = oncotreeCache.get(oncotreeCode);
+      }
+      switch (fieldName) {
+        case "cancerType":
+          return cachedData?.mainType;
+        case "cancerTypeDetailed":
+          return cachedData?.name;
+      }
+    }
+    switch (fieldName) {
+      case "recipe":
+        const cmoSampleIds =
+          parent.hasMetadataSampleMetadata?.[0]?.cmoSampleIdFields;
+        try {
+          const cmoSampleIdsObj = JSON.parse(cmoSampleIds);
+          return cmoSampleIdsObj["recipe"];
+        } catch {
+          return null;
+        }
+      case "dmpPatientId":
+        return parent.patientsHasSample?.[0]?.patientAliasesIsAlias?.find(
+          (patientAlias) => patientAlias.namespace === "dmpId"
+        )?.value;
+    }
+  },
+  Cohort: (parent, fieldName, _context) => {
+    switch (fieldName) {
+      case "totalSampleCount":
+        return parent.hasCohortSampleSamplesConnection?.totalCount;
+      case "smileSampleIds":
+        return parent.hasCohortSampleSamples?.map((s) => s.smileSampleId);
+      case "billed":
+        const samples = parent.hasCohortSampleSamples;
+        const allSamplesBilled =
+          samples?.length > 0 &&
+          samples.every((sample) => sample.hasTempoTempos?.[0]?.billed);
+        return allSamplesBilled ? "Yes" : "No";
+      case "initialCohortDeliveryDate":
+        return parent.hasCohortCompleteCohortCompletes?.slice(-1)[0]?.date;
+      case "endUsers":
+        return parent.hasCohortCompleteCohortCompletes?.[0]?.endUsers;
+      case "pmUsers":
+        return parent.hasCohortCompleteCohortCompletes?.[0]?.pmUsers;
+      case "projectTitle":
+        return parent.hasCohortCompleteCohortCompletes?.[0]?.projectTitle;
+      case "projectSubtitle":
+        return parent.hasCohortCompleteCohortCompletes?.[0]?.projectSubtitle;
+      case "status":
+        return parent.hasCohortCompleteCohortCompletes?.[0]?.status;
+      case "type":
+        return parent.hasCohortCompleteCohortCompletes?.[0]?.type;
+    }
+  },
+};
+
+export function sortArrayByNestedField(
+  arr: any[],
+  nodeLabel: keyof typeof nestedValueGetters,
+  fieldName: string,
+  sortOrder: SortDirection,
+  context?: ApolloServerContext
+) {
+  arr.sort((objA, objB) => {
+    let a = nestedValueGetters[nodeLabel](objA, fieldName, context);
+    let b = nestedValueGetters[nodeLabel](objB, fieldName, context);
+
+    if (a === null || a === undefined) return 1;
+    if (b === null || b === undefined) return -1;
+
+    if (Array.isArray(a)) a = a.join(", ");
+    if (Array.isArray(b)) b = b.join(", ");
+
+    if (typeof a === "number" && typeof b === "number") {
+      return sortOrder === "ASC" ? a - b : b - a;
+    } else if (typeof a === "string" && typeof b === "string") {
+      return sortOrder === "ASC"
+        ? a.localeCompare(b, "en", { sensitivity: "base" })
+        : b.localeCompare(a, "en", { sensitivity: "base" });
+    } else {
+      return 0;
+    }
+  });
+}
```

Comment:
This sorting function is almost entirely a copy-and-paste from its old location (`graphql-server/src/schemas/neo4j.ts`).